napari/layers/utils/color_manager.py

```diff
@@ -0,0 +1,521 @@
+from copy import deepcopy
+from typing import Any, Dict, Optional, Tuple, Union
+
+import numpy as np
+from pydantic import root_validator, validator
+
+from ...utils.colormaps import Colormap
+from ...utils.colormaps.categorical_colormap import CategoricalColormap
+from ...utils.colormaps.colormap_utils import ensure_colormap
+from ...utils.events import EventedModel
+from ...utils.events.custom_types import Array
+from ._color_manager_constants import ColorMode
+from .color_manager_utils import (
+    _validate_colormap_mode,
+    _validate_cycle_mode,
+    guess_continuous,
+    is_color_mapped,
+)
+from .color_transformations import (
+    ColorType,
+    normalize_and_broadcast_colors,
+    transform_color,
+    transform_color_with_defaults,
+)
+
+
+class ColorProperties(EventedModel):
```

Comment:
Can you copy over the relevant doc string from below here, you basically have all the info there, but would be nice to see it again.